From d393e4a607d015303f43b495350d499fecbbe43f Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Mon, 30 Mar 2020 17:29:55 -0700
Subject: [PATCH 1298/1852] phy: zynqmp: check the config correctly

commit a3dc4a82958814670cb831e53b525b2c858be69f from
https://github.com/Xilinx/linux-xlnx.git

Correct the config check, otherwise the module build fails as below.

  CC [M]  drivers/phy/phy-zynqmp.o
drivers/phy/phy-zynqmp.c:309:5: error: redefinition of 'xpsgtr_override_deemph'
  309 | int xpsgtr_override_deemph(struct phy *phy, u8 plvl, u8 vlvl)
      |     ^~~~~~~~~~~~~~~~~~~~~~
In file included from drivers/phy/phy-zynqmp.c:24:
./include/linux/phy/phy-zynqmp.h:25:19: note: previous definition of 'xpsgtr_override_deemph' was here
   25 | static inline int xpsgtr_override_deemph(struct phy *base, u8 plvl, u8 vlvl)
      |                   ^~~~~~~~~~~~~~~~~~~~~~
drivers/phy/phy-zynqmp.c:326:5: error: redefinition of 'xpsgtr_margining_factor'
  326 | int xpsgtr_margining_factor(struct phy *phy, u8 plvl, u8 vlvl)
      |     ^~~~~~~~~~~~~~~~~~~~~~~
In file included from drivers/phy/phy-zynqmp.c:24:
./include/linux/phy/phy-zynqmp.h:30:19: note: previous definition of 'xpsgtr_margining_factor' was here
   30 | static inline int xpsgtr_margining_factor(struct phy *base, u8 plvl, u8 vlvl)
      |                   ^~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Cc: Anurag Kumar Vulisha <anurag.kumar.vulisha@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 include/linux/phy/phy-zynqmp.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/phy/phy-zynqmp.h b/include/linux/phy/phy-zynqmp.h
index 81fdfe0391d1..d4185461abfe 100644
--- a/include/linux/phy/phy-zynqmp.h
+++ b/include/linux/phy/phy-zynqmp.h
@@ -13,7 +13,7 @@
 
 struct phy;
 
-#if defined(CONFIG_PHY_XILINX_ZYNQMP)
+#if IS_ENABLED(CONFIG_PHY_XILINX_ZYNQMP)
 
 extern int xpsgtr_override_deemph(struct phy *phy, u8 plvl, u8 vlvl);
 extern int xpsgtr_margining_factor(struct phy *phy, u8 plvl, u8 vlvl);
-- 
2.31.1

